src/gui/src/displayControls.cpp

```diff
@@ -155,10 +155,84 @@ void DisplayColorDialog::rejectDialog()
   reject();
 }
 
+DisplayControlModel::DisplayControlModel(QWidget* parent) :
+  QStandardItemModel(0, 4, parent)
+{
+}
+
+QVariant DisplayControlModel::data(const QModelIndex& index, int role) const
+{
+  if (role == Qt::ToolTipRole) {
+    QStandardItem* item = itemFromIndex(index);
+    QVariant data = item->data(Qt::UserRole);
+    if (data.isValid()) {
+      odb::dbTechLayer* layer = data.value<odb::dbTechLayer*>();
+      auto selected = Gui::get()->makeSelected(layer);
+      if (selected) {
+        auto props = selected.getProperties();
+        const QString micron = "\u03BC";
+
+        // provide tooltip with layer information
+        QString information;
+
+        auto add_prop = [props](const std::string& prop, QString& info) -> bool {
+          auto prop_find = std::find_if(props.begin(), props.end(), [prop](const auto& p) {
+            return p.name == prop;
+          });
+          if (prop_find == props.end()) {
+            return false;
+          }
+          info += "\n" + QString::fromStdString(prop) + ": ";
+          if (auto v = std::any_cast<const char*>(&(*prop_find).value)) {
+            info += QString(*v);
+          } else if (auto v = std::any_cast<const std::string>(&(*prop_find).value)) {
+            info += QString::fromStdString(*v);
+          } else if (auto v = std::any_cast<int>(&(*prop_find).value)) {
+            info += QString::number(*v);
+          } else if (auto v = std::any_cast<unsigned int>(&(*prop_find).value)) {
+            info += QString::number(*v);
+          } else if (auto v = std::any_cast<double>(&(*prop_find).value)) {
+            info += QString::number(*v);
+          } else if (auto v = std::any_cast<float>(&(*prop_find).value)) {
+            info += QString::number(*v);
+          } else {
+            info += "<unknown>";
+          }
```

Comment:
This looks to repeat what is done in convertAnyToQString.   Can that be shared?  Perhaps it should become a method on Property.

Reply:
like a toString? for the property, that would be doable